src/test/java/io/xstefank/wildfly/bot/utils/Mockable.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * @brief This class serves as base for grouping Mocking functionality
+ * for certain logical domains, such as {@link org.kohsuke.github.GHRepository}
+ * or {@link org.kohsuke.github.GHPullRequest}.
+ * Where you would create a {@code Mockable} for your specific domain and can
+ * chain other {@code Mockable}-s together, when mocking. See usage in tests.
+ *
+ * You can add your implementation of Mockable into {@code requiredMockables} Set.
+ * After all registered {@code Mockable}-s (by {@code mockNext}) have been executed
+ * and no instances from {@code requiredMockables} have been found, then default implementation
+ * will be executed. This is the implementation you add to the {@code requiredMockables}.
+ *
+ * Note: Only one {@code Mockable} per domain is allowed to be mocked.
+ *
+ * Parallel execution is not supported as we are relaying static variables.
+ */
+package io.xstefank.wildfly.bot.utils;
+
+import com.thoughtworks.xstream.InitializationException;
+import io.quarkiverse.githubapp.testing.dsl.GitHubMockContext;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static io.xstefank.wildfly.bot.utils.TestConstants.VALID_PR_TEMPLATE_JSON;
+
+public abstract class Mockable {
+
+    protected Mockable previousMock;
+
+    private static final Set<Mockable> requiredMockables;
+    static {
```

Comment:
This section doesn't seem just right. Perhaps I can convert this into an injectable bean, which could based on a new annotation pick up all Mockable-s and fill the set this way. It would also not rely on static variables and context.
However, that might be a bit too overkill for testing purposes and this might be more readable and understandable for future reference.

Let me know, what you think @xstefank

Reply:
TBH, not sure if there should be a default at all. If this is used everywhere, it can be in the constructor.

Reply:
usually a new check or functionality always calls some new API. Thus a default mock needs to be there, otherwise it would throw many exceptions.

Example, adding check for config file always retrieves a file, thus every test will try to retrieve some file. We mock that no file is there to retrieve instead of NPE from mockito.

Reply:
It's static, as this method is called in tests annotated with `BeforeAll`, which has to be on static method. Thus static context is required.